exporter/utils/node_details.py

```diff
@@ -0,0 +1,64 @@
+from exporter.osclient import get_ironic_client, session_adapter
+
+FREEPOOL_AGGREGATE_ID = 1
+
+
+def get_nodes(detail=False):
+    """Return list of ironic client node objects."""
+    ironic_client = get_ironic_client()
+    return ironic_client.node.list(detail=detail)
+
+
+def add_project_names(nodes):
+    """Add project names to list of ironic client node objects."""
+    nova_api = session_adapter('compute')
+    keystone_api = session_adapter('identity')
+
+    aggregates = nova_api.get('os-aggregates').json()['aggregates']
+    projects = keystone_api.get('v3/projects').json()['projects']
+
+    project_names = {p['id']: p['name'] for p in projects}
+    reservations = dict()
+
+    for agg in aggregates:
+        # Ignore projects in freepool
+        if agg['id'] == FREEPOOL_AGGREGATE_ID or not agg['hosts']:
+            continue
+
+        project_id = agg['metadata']['blazar:owner']
+
+        for node_id in agg['hosts']:
+            reservations[node_id] = project_names[project_id]
+
+    for node in nodes:
+        if node.uuid in reservations:
+            setattr(node, 'project_name', reservations[node.uuid])
+        else:
+            setattr(node, 'project_name', None)
+
+
+def add_node_type(nodes):
+    """Add node_type to list of ironic client node objects."""
+    blazar_api = session_adapter('reservation')
+    hosts = blazar_api.get('os-hosts?detail=True').json()['hosts']
+
+    hosts_by_node = {
+        h['hypervisor_hostname']:
+            {'node_type': h['node_type'], 'gpu': h['gpu.gpu']}
```

Comment:
Do we need the `gpu` property here? I think in Prometheus we can probably just use a fuzzy match on the node_type to catch the GPU-ness of a node. I know this code wasn't changed from before, just commenting.